### PR TITLE
Update Entity Converting Tests to NPC Factory

### PR DESCRIPTION
**Pull request by Jul-Wong**
# Description
Adds tests to NPC factory and completes #457. Has no change on game functionality, only changing testing classes mainly

### CODE REVIEW

**Review by theboah — CHANGES_REQUESTED** [review]
Please resolve conflicts ready for merge then will recheck builds and runs. Tests look valid otherwise.

**Review by theboah — APPROVED** [review]

File: source/core/src/test/com/csse3200/game/minigames/maze/entities/factories/MazeNPCFactoryTest.java [hunk 1]
```diff
@@ -30,7 +30,6 @@
 import static org.junit.jupiter.api.Assertions.assertNotNull;
 import static org.mockito.Mockito.*;
 
-@ExtendWith(GameExtension.class)
```
Comment (mitsudz) [inline source/core/src/test/com/csse3200/game/minigames/maze/entities/factories/MazeNPCFactoryTest.java:33]: If you get rid of this the test won't actually run when building.

File: source/core/src/test/com/csse3200/game/minigames/maze/entities/factories/AnglerFishTest.java [hunk 2]
```diff
@@ -95,6 +95,7 @@ void testMazeCombatStatsComponent() {
         assertNotNull(combatStats, "AnglerFish should have a MazeCombatStatsComponent");
         assertEquals(100, combatStats.getHealth(), "Health should be set from config");
         assertEquals(20, combatStats.getBaseAttack(), "Base attack should be set from config");
+
```
Comment (mitsudz) [inline source/core/src/test/com/csse3200/game/minigames/maze/entities/factories/AnglerFishTest.java:98]: ?

File: source/core/src/test/com/csse3200/game/minigames/maze/components/MazeTouchAttackComponentTest.java [hunk 3]
```diff
@@ -125,5 +125,6 @@ private Entity createTarget(short layer) {
         return new Entity()
                 .addComponent(new MazeCombatStatsComponent(10, 10, 0))
                 .addComponent(new HitboxComponent().setLayer(layer));
+
```
Comment (mitsudz) [inline source/core/src/test/com/csse3200/game/minigames/maze/components/MazeTouchAttackComponentTest.java:128]: ?

File: source/core/src/test/com/csse3200/game/minigames/maze/components/MazeCombatStatsComponentTest.java [hunk 4]
```diff
@@ -56,4 +56,5 @@ void testSetBaseAttack() {
         player.setBaseAttack(-5); // Should not allow negative attack values
         assertEquals(20, player.getBaseAttack(), "Base attack should remain 20 after trying to set a negative value");
     }
+
```
Comment (mitsudz) [inline source/core/src/test/com/csse3200/game/minigames/maze/components/MazeCombatStatsComponentTest.java:59]: ?

File: source/core/src/test/com/csse3200/game/entities/factories/NPCFactoryTest.java [hunk 5]
```diff
@@ -114,8 +128,285 @@ static void setup() {
...
+    void TestMonkeyHasCorrectBaseHint() {
+        String[][] baseHint = configs.friendlyMonkey.getBaseHint();
+        assertNotNull(baseHint);
+        Assertions.assertArrayEquals(new String[][]{{"Welcome to Animal Kingdom!","I am Max the Monkey."}},
+                baseHint);
+    }
+
+    /**
+     * Tests that the friendly monkey has a wait animation.
+     */
```
Comment (mitsudz) [inline source/core/src/test/com/csse3200/game/entities/factories/NPCFactoryTest.java:382]: Why is this test commented out? Please either make it work with the current game or remove it.

File: source/core/src/test/com/csse3200/game/entities/factories/NPCFactoryTest.java [hunk 6]
```diff
@@ -114,8 +128,285 @@ static void setup() {
...
+    @Test
+    void TestFrogHasCorrectBaseHint() {
+        String[][] baseHint = configs.friendlyFrog.getBaseHint();
+        assertNotNull(baseHint);
+        Assertions.assertArrayEquals(new String[][]{{"Welcome to Animal Kingdom!","I am Fred the Frog."}},
+                baseHint);
+    }
+
+    /**
+     * Tests that the friendly frog has a wait animation.
```
Comment (mitsudz) [inline source/core/src/test/com/csse3200/game/entities/factories/NPCFactoryTest.java:300]: Why is this test commented out? Please either make it work with the current game or remove it.

File: source/core/src/test/com/csse3200/game/entities/factories/NPCFactoryTest.java [hunk 7]
```diff
@@ -114,8 +128,285 @@ static void setup() {
...
+        String[][] baseHint = configs.friendlyChicken.getBaseHint();
+        assertNotNull(baseHint);
+        Assertions.assertArrayEquals(new String[][]{{"Welcome to Animal Kingdom!","I am Charlie the Chicken."}},
+                baseHint);
+    }
+
+    /**
+     * Tests that the friendly chicken has a wait animation.
+     */
+//    @Test
```
Comment (mitsudz) [inline source/core/src/test/com/csse3200/game/entities/factories/NPCFactoryTest.java:219]: Why is this test commented out? Please either make it work with the current game or remove it.

File: source/core/src/test/com/csse3200/game/entities/factories/NPCFactoryTest.java [hunk 8]
```diff
@@ -114,8 +128,285 @@ static void setup() {
...
+        String[][] testHintText = new String[][]{
+                {"Welcome to Animal Kingdom!", "I am Lenny the Lion.", "/cWhich tip do you wanna hear about?/s01What do potions do???/s02How to beat the final boss/s03Nothing. Bye"},
+                {"Potions heals you by (n) HP!", "I hope this helped."},
+                {"Final boss?? That Kangaroo??", "idk"},
+                {"Good luck!"}
+        };
+        NPCFactory.initiateDialogue(testSoundPath, testHintText);
+        verify(dialogueBoxService).updateText(testHintText, -2);
+
+//        for (String soundPath : testSoundPath) {
```
Comment (mitsudz) [inline source/core/src/test/com/csse3200/game/entities/factories/NPCFactoryTest.java:159]: Why is this test commented out? Please either make it work with the current game or remove it.

File: source/core/src/main/com/csse3200/game/screens/MazeGameScreen.java [hunk 9]
```diff
@@ -213,6 +212,7 @@ private void unloadAssets() {
      */
     private void createUI() {
         logger.debug("Creating ui");
+
```
Comment (mitsudz) [inline source/core/src/main/com/csse3200/game/screens/MazeGameScreen.java:215]: ?

File: source/core/src/main/com/csse3200/game/screens/MazeGameScreen.java [hunk 10]
```diff
@@ -102,7 +102,6 @@ public MazeGameScreen(GdxGame game, Screen screen, ServiceContainer container) {
         renderer.getDebug().renderPhysicsWorld(physicsEngine.getWorld());
 
         LightingEngine lightingEngine = new LightingEngine(physicsEngine.getWorld(), camComponent.getCamera());
-
```
Comment (mitsudz) [inline source/core/src/main/com/csse3200/game/screens/MazeGameScreen.java:105]: ?

File: source/core/src/main/com/csse3200/game/screens/EndMiniGameScreen.java [hunk 11]
```diff
@@ -123,6 +123,7 @@ public void render(float delta) {
     }
 
     /**
+
```
Comment (mitsudz) [inline source/core/src/main/com/csse3200/game/screens/EndMiniGameScreen.java:126]: ?

File: source/core/src/main/com/csse3200/game/minigames/maze/entities/mazenpc/Jellyfish.java [hunk 12]
```diff
@@ -42,6 +42,7 @@ public Jellyfish(MazeEntityConfig config) {
         this.addComponent(new MazeCombatStatsComponent(config.health, config.baseAttack, config.speed))
                 .addComponent(animator)
                 .addComponent(new MazeEntityAnimationController())
+
```
Comment (mitsudz) [inline source/core/src/main/com/csse3200/game/minigames/maze/entities/mazenpc/Jellyfish.java:45]: ?

File: source/core/src/main/com/csse3200/game/minigames/maze/entities/mazenpc/FishEgg.java [hunk 13]
```diff
@@ -19,6 +19,7 @@
  * It does not actively chase or attack the player.
  */
 public class FishEgg extends Entity {
+
```
Comment (mitsudz) [inline source/core/src/main/com/csse3200/game/minigames/maze/entities/mazenpc/FishEgg.java:22]: ?

File: source/core/src/main/com/csse3200/game/minigames/maze/entities/MazePlayer.java [hunk 14]
```diff
@@ -66,6 +66,7 @@ public MazePlayer(MazePlayerConfig stats, MazeGameArea gameArea) {
                 .addComponent(new MazeDifficultyIncrease(gameArea))
                 .addComponent(new ParticleEffectComponent("images/minigames/trail.p"));
 
+
```
Comment (mitsudz) [inline source/core/src/main/com/csse3200/game/minigames/maze/entities/MazePlayer.java:69]: ?

File: source/core/src/main/com/csse3200/game/minigames/maze/components/MazeCombatStatsComponent.java [hunk 15]
```diff
@@ -92,6 +92,7 @@ public void setBaseAttack(int attack) {
         }
     }
 
+
```
Comment (mitsudz) [inline source/core/src/main/com/csse3200/game/minigames/maze/components/MazeCombatStatsComponent.java:95]: ?

File: source/core/src/main/com/csse3200/game/minigames/maze/areas/MazeGameArea.java [hunk 16]
```diff
@@ -286,6 +288,7 @@ private void spawnFishEggs() {
     }
 
     /**
+     * Playes backgroun music for the maze mini-game
```
Comment (mitsudz) [inline source/core/src/main/com/csse3200/game/minigames/maze/areas/MazeGameArea.java:291]: ?

File: source/core/src/main/com/csse3200/game/minigames/maze/areas/MazeGameArea.java [hunk 17]
```diff
@@ -271,7 +271,9 @@ private void spawnEels() {
         for (int i = 0; i < MazeGameArea.NUM_EELS; i++) {
             Entity eel = MazeNPCFactory.createEel(player);
             spawnEntityAt(eel, getSimpleStartLocation(3f), true, true);
+
             getEnemies(Entity.EnemyType.MAZE_EEL).add(eel);
+
```
Comment (mitsudz) [inline source/core/src/main/com/csse3200/game/minigames/maze/areas/MazeGameArea.java:276]: ?

File: source/core/src/main/com/csse3200/game/minigames/maze/areas/MazeGameArea.java [hunk 18]
```diff
@@ -271,7 +271,9 @@ private void spawnEels() {
         for (int i = 0; i < MazeGameArea.NUM_EELS; i++) {
             Entity eel = MazeNPCFactory.createEel(player);
             spawnEntityAt(eel, getSimpleStartLocation(3f), true, true);
+
```
Comment (mitsudz) [inline source/core/src/main/com/csse3200/game/minigames/maze/areas/MazeGameArea.java:274]: ?

File: source/core/src/main/com/csse3200/game/minigames/maze/Maze.java [hunk 19]
```diff
@@ -74,6 +74,7 @@ public boolean isWall(int x, int y, GridPoint2 direction) {
         return isWall(new GridPoint2(x, y), direction);
     }
 
+
```
Comment (mitsudz) [inline source/core/src/main/com/csse3200/game/minigames/maze/Maze.java:77]: ?

**Review by mitsudz — CHANGES_REQUESTED** [review]
Seems to be a lot of random changes to do with spacing that don't improve styling. Also a bunch of commented out tests which should either be removed or made to work. Please look through this pr yourself - there is also a error file you're trying to send to main, and an extra 2024-studio-2 directory which shouldn't be there. I'm also unsure why you're deleting the .github/pull_request_template.md file.

**Pull request closed without merging**